python/src/shaprpy/explanation.py

```diff
@@ -0,0 +1,242 @@
+"""
+Shapr explanation class for exploring Shapley value results.
+"""
+from encodings.punycode import digits
+from rpy2 import robjects
+from shaprpy._rutils import _importr
+
+
+class Shapr:
+    """
+    A class to hold and explore Shapley value explanations using R functions.
+
+    This class wraps the explanation dictionary and R object returned by the
+    explain() function, providing direct access to R's shapr functions for
+    exploration and analysis.
+
+    Parameters
+    ----------
+    explanation_dict : dict
+        The explanation dictionary returned by explain().
+    r_object : R object
+        The original R shapr object used for all R function calls.
+    """
+
+    def __init__(self, explanation_dict, r_object):
+        """
+        Initialize the Shapr explanation object.
+
+        Parameters
+        ----------
+        explanation_dict : dict
+            The explanation dictionary containing Shapley values and metadata.
+        r_object : R object
+            The original R shapr object for accessing R-specific functionality.
+        """
+        self._explanation_dict = explanation_dict
+        self._r_object = r_object
+
+    def get_explanation_dict(self):
+        """Get the explanation dictionary.
+
+        Returns the original Python dictionary that was returned by explain()
+        before being wrapped in this class. This provides direct access to
+        all the Python-converted components of the explanation.
+
+        Returns
+        -------
+        dict
+            The explanation dictionary containing keys like 'shapley_values_est',
+            'internal', 'timing', etc.
+
+        Examples
+        --------
+        >>> explanation = shaprpy.explain(model, X_test, X_train)
+        >>> exp_dict = explanation.get_explanation_dict()
+        >>> shapley_values = exp_dict['shapley_values_est']
+        """
+        return self._explanation_dict
+
+    def get_r_object(self):
+        """
+        Get the original R shapr object.
+
+        This allows direct access to R functions like print.shapr(),
+        plotting functions, etc.
+
+        Returns
+        -------
+        R object
+            The original R shapr object.
+        """
+        return self._r_object
+
+
+
+    def get_results(self, what=None):
+        """
+        Extract components from the Shapr explanation object using R's get_results function.
+
+        Parameters
+        ----------
+        what : str or list of str, optional
+            Component(s) to extract. Available options include:
+            "calling_function", "approach", "shapley_est", "shapley_sd", "pred_explain",
```

Comment:
```suggestion
            "calling_function", "proglang", "approach", "shapley_est", "shapley_sd", "pred_explain",
```

Ser ut "proglang" mangler. Den er lov i R i alle fall.